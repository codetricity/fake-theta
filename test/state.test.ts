import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('POST /osc/state', () => {
  it('should respond with state', async () => {
    const res = await request().post('/osc/state').expect(200);
    expect(res.text).toMatchSnapshot();
  });
});

describe('THETA X POST /osc/state', () => {
  it('should respond with state', async () => {
    const res = await request()
      .post('/osc/state')
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });
});

describe('THETA Z1 POST /osc/state', () => {
  it('should respond with state', async () => {
    const res = await request()
      .post('/osc/state')
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });
});

describe('Unknown Type POST /osc/state', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/state')
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
