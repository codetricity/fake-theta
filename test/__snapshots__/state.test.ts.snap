// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/state should respond with state 1`] = `
{
  "fingerprint": "FIG_0001",
  "state": {
    "_apiVersion": 2,
    "_batteryInsert": true,
    "_batteryState": "disconnect",
    "_cameraError": [],
    "_captureStatus": "idle",
    "_capturedPictures": 0,
    "_currentMicrophone": "Internal",
    "_currentStorage": "IN",
    "_function": "normal",
    "_latestFileUrl": "http://192.168.1.1/files/100RICOH/R0010015.JPG",
    "_mySettingChanged": false,
    "_pluginRunning": false,
    "_pluginWebServer": false,
    "_recordableTime": 0,
    "_recordedTime": 0,
    "_storageID": "90014a68423861503e030277e0c2b500",
    "batteryLevel": 0.33,
    "storageUri": "http://192.168.1.1/files/",
  },
}
`;

exports[`THETA X POST /osc/state should respond with state 1`] = `
{
  "fingerprint": "FIG_0001",
  "state": {
    "_apiVersion": 2,
    "_batteryInsert": true,
    "_batteryState": "disconnect",
    "_cameraError": [],
    "_captureStatus": "idle",
    "_capturedPictures": 0,
    "_currentMicrophone": "Internal",
    "_currentStorage": "IN",
    "_function": "normal",
    "_latestFileUrl": "http://192.168.1.1/files/100RICOH/R0010015.JPG",
    "_mySettingChanged": false,
    "_pluginRunning": false,
    "_pluginWebServer": false,
    "_recordableTime": 0,
    "_recordedTime": 0,
    "_storageID": "90014a68423861503e030277e0c2b500",
    "batteryLevel": 0.33,
    "storageUri": "http://192.168.1.1/files/",
  },
}
`;

exports[`THETA Z1 POST /osc/state should respond with state 1`] = `
{
  "fingerprint": "FIG_0001",
  "state": {
    "_apiVersion": 2,
    "_batteryState": "disconnect",
    "_cameraError": [],
    "_captureStatus": "idle",
    "_capturedPictures": 0,
    "_compositeShootingElapsedTime": 0,
    "_function": "normal",
    "_latestFileUrl": "http://192.168.1.1/files/150100525831424d42075b53ce68c300/100RICOH/R0010015.JPG",
    "_mySettingChanged": false,
    "_pluginRunning": false,
    "_pluginWebServer": false,
    "_recordableTime": 0,
    "_recordedTime": 0,
    "batteryLevel": 0.33,
    "storageUri": "http://192.168.1.1/files/150100525831424d42075b53ce68c300/",
  },
}
`;
